metrics/release_test.go

```diff
@@ -17,239 +17,225 @@ limitations under the License.
+					releaseTotalOpts,
+					releaseTotalLabels,
+				))).To(Succeed())
 		})
 	})
 
-	Context("When RegisterCompletedRelease is called", func() {
-		BeforeAll(func() {
-			ReleaseAttemptRunningSeconds = prometheus.NewHistogram(
-				prometheus.HistogramOpts{
-					Name:    "release_attempt_running_seconds",
-					Help:    "Release durations from the moment the release resource was created til the release is marked as running",
-					Buckets: []float64{1, 5, 10, 30},
-				},
-			)
-			ReleaseDurationSeconds = prometheus.NewHistogramVec(
-				prometheus.HistogramOpts{
-					Name:    "release_attempt_duration_seconds",
-					Help:    "Release durations from the moment the release PipelineRun was created til the release is marked as finished",
-					Buckets: []float64{60, 600, 1800, 3600},
-				},
-				[]string{"reason", "strategy", "succeeded", "target"},
-			)
+	Context("When RegisterCompletedReleaseDeployment is called", func() {
+		var completionTime, startTime *metav1.Time
 
-			ReleaseConcurrentTotal = prometheus.NewGauge(
-				prometheus.GaugeOpts{
-					Name: "release_attempt_concurrent_requests",
-					Help: "Total number of concurrent release attempts",
-				},
-			)
-			metrics.Registry.MustRegister(ReleaseAttemptRunningSeconds, ReleaseDurationSeconds, ReleaseConcurrentTotal)
+		BeforeEach(func() {
+			initializeMetrics()
+
+			completionTime = &metav1.Time{}
+			startTime = &metav1.Time{Time: completionTime.Add(-60 * time.Second)}
+		})
+
+		It("decrements ReleaseConcurrentDeploymentsTotal", func() {
+			Expect(testutil.ToFloat64(ReleaseConcurrentDeploymentsTotal.WithLabelValues())).To(Equal(float64(0)))
+			RegisterCompletedReleaseDeployment(startTime, completionTime, "", "", "")
+			Expect(testutil.ToFloat64(ReleaseConcurrentDeploymentsTotal.WithLabelValues())).To(Equal(float64(-1)))
 		})
 
-		AfterAll(func() {
-			metrics.Registry.Unregister(ReleaseAttemptRunningSeconds)
-			metrics.Registry.Unregister(ReleaseDurationSeconds)
-			metrics.Registry.Unregister(ReleaseConcurrentTotal)
+		It("adds an observation to ReleaseDeploymentDurationSeconds", func() {
+			RegisterCompletedReleaseDeployment(startTime, completionTime,
+				releaseDeploymentDurationSecondsLabels[0],
+				releaseDeploymentDurationSecondsLabels[1],
+				releaseDeploymentDurationSecondsLabels[2],
+			)
+			Expect(testutil.CollectAndCompare(ReleaseDeploymentDurationSeconds,
+				newHistogramReader(
+					releaseDeploymentDurationSecondsOpts,
+					releaseDeploymentDurationSecondsLabels,
+					startTime, completionTime,
+				))).To(Succeed())
 		})
+	})
 
-		// Input seconds for duration of operations less or equal to the following buckets of 60, 600, 1800 and 3600 seconds
-		inputSeconds := []float64{30, 500, 1500, 3000}
-		elapsedSeconds := 0.0
-		labels := fmt.Sprintf(`reason="%s", strategy="%s", succeeded="true", target="%s",`,
-			validReleaseReason, strategy, defaultNamespace)
+	Context("When RegisterCompletedReleaseProcessing is called", func() {
```

Comment:
why is this in a different order than the functions are defined in release.go?

Reply:
My bad. Fixed.